orientdb/src/main/java/com/yahoo/ycsb/db/OrientDBClient.java

```diff
@@ -68,42 +76,70 @@ public void init() throws DBException {
     String user = props.getProperty(USER_PROPERTY, USER_PROPERTY_DEFAULT);
     String password = props.getProperty(PASSWORD_PROPERTY, PASSWORD_PROPERTY_DEFAULT);
     Boolean newdb = Boolean.parseBoolean(props.getProperty(NEWDB_PROPERTY, NEWDB_PROPERTY_DEFAULT));
+    String remoteStorageType = props.getProperty(STORAGE_TYPE_PROPERTY);
 
     if (url == null) {
       throw new DBException(String.format("Required property \"%s\" missing for OrientDBClient", URL_PROPERTY));
     }
 
-    try {
-      System.out.println("OrientDB loading database url = " + url);
-
-      OGlobalConfiguration.STORAGE_KEEP_OPEN.setValue(false);
-      db = new ODatabaseDocumentTx(url);
-      if (db.exists()) {
-        db.open(user, password);
-        if (newdb) {
-          System.out.println("OrientDB drop and recreate fresh db");
-          db.drop();
-          db.create();
-        }
-      } else {
-        System.out.println("OrientDB database not found, create fresh db");
-        db.create();
-      }
-
-      System.out.println("OrientDB connection created with " + url);
+    System.err.println("OrientDB loading database url = " + url);
 
-      dictionary = db.getMetadata().getIndexManager().getDictionary();
-      if (!db.getMetadata().getSchema().existsClass(CLASS))
-        db.getMetadata().getSchema().createClass(CLASS);
+    // If using a remote database, use the OServerAdmin interface to connect
+    if (url.startsWith(OEngineRemote.NAME)) {
+      isRemote = true;
+      if (remoteStorageType == null) {
+        throw new DBException("When connecting to a remote OrientDB instance, specify a database storage type (plocal or memory) with " + STORAGE_TYPE_PROPERTY);
+      }
 
-      db.declareIntent(new OIntentMassiveInsert());
+      try {
+        OServerAdmin server = new OServerAdmin(url).connect(user, password);
+        String dbpath = url.split(":", 2)[1];
+
+        if (server.existsDatabase()) {
+          if (newdb) {
+            System.err.println("OrientDB dropping and recreating fresh db on remote server.");
+            server.dropDatabase(remoteStorageType);
+            server.createDatabase(dbpath, ORIENTDB_DOCUMENT_TYPE, remoteStorageType);
+          }
+        } else {
+          System.err.println("OrientDB database not found, creating fresh db");
+          server.createDatabase(dbpath, ORIENTDB_DOCUMENT_TYPE, remoteStorageType);
+        }
 
-    } catch (Exception e1) {
-      System.err.println("Could not initialize OrientDB connection pool for Loader: " + e1.toString());
-      e1.printStackTrace();
-      return;
+        server.close();
+        db = new ODatabaseDocumentTx(url).open(user, password);
+      } catch (IOException | OException e) {
+        System.err.println(String.format("Error interfacing with %s", url));
+        e.printStackTrace();
```

Comment:
Yes, it should, thanks for catching this, it was not done correctly initially either but in both code paths throwing a DBException is the right way to go.